lib/web_ui/lib/src/engine/text/paragraph.dart

```diff
@@ -292,19 +292,21 @@ class EngineParagraph implements ui.Paragraph {
   /// - Paragraphs that have a non-null word-spacing.
   /// - Paragraphs with a background.
   bool get _drawOnCanvas {
+    if (_measurementResult.lines == null) {
+      return false;
+    }
+
     bool canDrawTextOnCanvas;
-    if (TextMeasurementService.enableExperimentalCanvasImplementation) {
-      canDrawTextOnCanvas = _measurementResult.lines != null;
+    if (_measurementService is CanvasTextMeasurementService) {
```

Comment:
`is` checks are expensive 🤔 But I guess this will be going away eventually, right?

Reply:
Hmm I didn't know that. I'm assuming that's because it has to check for subclasses, etc?

Maybe I'll add a boolean field `TextMeasurementService.isCanvas`. What do you think?